### PR TITLE
TEST: Create following engines in the main thread

### LINKED ISSUE #33344: [CI] LuceneChangesSnapshotTests#testUpdateAndReadChangesConcurrently fails
`LuceneChangesSnapshotTests#testUpdateAndReadChangesConcurrently` fails occasionally on master and 6x. I reproduce locally with the following seed (the failure doesn't occur on each run though but I was able to reproduce after a few ones):
```
https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.x+release-tests/1005/console

./gradlew :server:test \
  -Dtests.seed=95ED5AB5FEF63FF6 \
  -Dtests.class=org.elasticsearch.index.engine.LuceneChangesSnapshotTests \
  -Dtests.method="testUpdateAndReadChangesConcurrently" \
  -Dtests.security.manager=true \
  -Dbuild.snapshot=false \
  -Dtests.jvm.argline="-Dbuild.snapshot=false" \
  -Dtests.locale=bg \
  -Dtests.timezone=Pacific/Enderbury \
  -Dcompiler.java=10 \
  -Druntime.java=8
```

### PR DESCRIPTION
**Pull request by dnhatn**
There are two races in the `testUpdateAndReadChangesConcurrently` if the following engines are created in the worker threads. We fixed the translog issue in #33352, but there is still another race with `createStore`.

This commit ensures that we create all following engines in the main thread.

Relates #33352
Closes #33344

### CODE REVIEW

**Review by original-brownbear — APPROVED** [review]
LGTM :)

File: server/src/test/java/org/elasticsearch/index/engine/LuceneChangesSnapshotTests.java [hunk 1]
```diff
@@ -202,7 +200,7 @@ public void testUpdateAndReadChangesConcurrently() throws Exception {
         CountDownLatch readyLatch = new CountDownLatch(followers.length + 1);
```
Comment (original-brownbear) [inline server/src/test/java/org/elasticsearch/index/engine/LuceneChangesSnapshotTests.java]: NIT: I guess we could just set this to count `1` now and only keep the count down in the main thread? (the countdowns in the `run` method seem redundant now that it's not doing anything before invoking `countDown` right?)

Comment (dnhatn) [reply server/src/test/java/org/elasticsearch/index/engine/LuceneChangesSnapshotTests.java]: We use "readLatch" to ensure that all threads are ready (i.e. scheduled) before testing.

Comment (original-brownbear) [reply server/src/test/java/org/elasticsearch/index/engine/LuceneChangesSnapshotTests.java]: Ah right, that makes sense -> sorry for the noise :)

Comment (dnhatn) [reply server/src/test/java/org/elasticsearch/index/engine/LuceneChangesSnapshotTests.java]: no worries @original-brownbear :)

**Comment by dnhatn** [comment]
@original-brownbear thanks for reviewing.

**Pull request merged**